packages/mdc-floating-label/_mixins.scss

```diff
@@ -0,0 +1,87 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+@import "./variables";
+
+// Public
+
+@mixin mdc-floating-label-color($color) {
+  .mdc-floating-label {
+    @include mdc-theme-prop(color, $color);
+  }
+}
+
+// Private
+
+@mixin mdc-floating-label-invalid-shake-keyframes_($modifier, $positionY, $positionX: 0%, $scale: .75) {
+  @keyframes invalid-shake-float-above-#{$modifier} {
+    0% {
+      transform: translateX(calc(0 - #{$positionX})) translateY(-#{$positionY}) scale(#{$scale});
+    }
+
+    33% {
+      animation-timing-function: cubic-bezier(.5, 0, .701732, .495819);
+      transform: translateX(calc(4% - #{$positionX})) translateY(-#{$positionY}) scale(#{$scale});
+    }
+
+    66% {
+      animation-timing-function: cubic-bezier(.302435, .381352, .55, .956352);
+      transform: translateX(calc(-4% - #{$positionX})) translateY(-#{$positionY}) scale(#{$scale});
+    }
+
+    100% {
+      transform: translateX(calc(0 - #{$positionX})) translateY(-#{$positionY}) scale(#{$scale});
+    }
+  }
+}
+
+@include mdc-floating-label-invalid-shake-keyframes_(standard, $mdc-floating-label-position-y);
+@include mdc-floating-label-invalid-shake-keyframes_(box, $mdc-floating-label-box-position-y);
+@include mdc-floating-label-invalid-shake-keyframes_(box-dense, $mdc-floating-label-box-dense-position-y, 0%, $mdc-floating-label-dense-scale);
+@include mdc-floating-label-invalid-shake-keyframes_(outlined, $mdc-floating-label-outlined-position-y);
+@include mdc-floating-label-invalid-shake-keyframes_(outlined-dense, $mdc-floating-label-outlined-dense-position-y, 0%, $mdc-floating-label-dense-scale);
+@include mdc-floating-label-invalid-shake-keyframes_(outlined-leading-icon, $mdc-floating-label-outlined-position-y, $mdc-floating-label-outlined-with-leading-icon-position-x);
+@include mdc-floating-label-invalid-shake-keyframes_(outlined-leading-icon-dense, $mdc-floating-label-outlined-dense-position-y, $mdc-floating-label-outlined-dense-with-leading-icon-position-x, $mdc-floating-label-dense-scale);
+@include mdc-floating-label-invalid-shake-keyframes_(textarea, $mdc-floating-label-textarea-position-y, 0%, $mdc-floating-label-textarea-scale);
+
+@mixin mdc-floating-label-position_($positionY, $positionX: 0%, $scale: .75) {
+  @if $positionX == 0 {
+    .mdc-floating-label--float-above {
+      transform: translateY(-1 * $positionY) scale($scale);
+    }
+  } @else {
+    .mdc-floating-label--float-above {
+      transform: translateY(-1 * $positionY) translateX(-1 * $positionX) scale($scale);
+
+      @include mdc-rtl {
+        transform: translateY(-1 * $positionY) translateX($positionX) scale($scale);
+      }
+    }
+  }
+}
+
+@mixin mdc-floating-label-shake_($modifier) {
```

Comment:
This is used heavily in text-field, so I think this should actually be made public, along with reasons made in https://github.com/material-components/material-components-web/pull/2237/files#r167720790